src/include/duckdb/optimizer/join_order/join_relation.hpp

```diff
@@ -51,13 +42,13 @@ class JoinRelationSetManager {
 
 public:
 	//! Create or get a JoinRelationSet from a single node with the given index
-	JoinRelationSet &GetJoinRelation(idx_t index);
+	optional_ptr<JoinRelationSet> GetJoinRelation(idx_t index);
 	//! Create or get a JoinRelationSet from a set of relation bindings
-	JoinRelationSet &GetJoinRelation(const unordered_set<idx_t> &bindings);
+	optional_ptr<JoinRelationSet> GetJoinRelation(const unordered_set<idx_t> &bindings);
 	//! Create or get a JoinRelationSet from a (sorted, duplicate-free!) list of relations
-	JoinRelationSet &GetJoinRelation(unsafe_unique_array<idx_t> relations, idx_t count);
+	optional_ptr<JoinRelationSet> GetJoinRelation(unsafe_unique_array<idx_t> relations, idx_t count);
 	//! Union two sets of relations together and create a new relation set
-	JoinRelationSet &Union(JoinRelationSet &left, JoinRelationSet &right);
+	optional_ptr<JoinRelationSet> Union(optional_ptr<JoinRelationSet> left, optional_ptr<JoinRelationSet> right);
```

Comment:
Same thing here, references are preferred if they are always there. If they are not always there, and they are really optional, feel free to ignore these comments :)